File: python/examples/mode.py
import sys, random

sys.path.insert(0, '../lib/')
from mean_averages import MeanAverages

if __name__ == "__main__":
    sample_data = open('../../sample_data', 'r')

    sample_index = 1
    for line in sample_data:
        if not line or line.isspace():
            continue

        print('Sample ', sample_index)
        
        sample = [ float(data) for data in line.split(',') ]
        print('Sample Data: ', sample)
        print('Mode: ', MeanAverages.mode(sample))

        sample_index += 1
        print()
        print()
    
    sample_data.close()

    sample = [random.randint(1, 20) for x in range(1, 25)]
    print('Sample ', sample_index)
    print('Sample Data: ', sample)
    print('Mode: ', MeanAverages.mode(sample))

File: php/examples/mode.php
<?php

require_once('../lib/mean_averages.php');

function calculateMean($sampleIndex, $sample) {
    $average = ClounceMath\MeanAverages::mode($sample);
    
    print("Sample $sampleIndex" . PHP_EOL);
    print('Sample Data: ' . implode(', ', $sample) . PHP_EOL);
    print("Mode: " . implode(', ', $average) . PHP_EOL);
    print(PHP_EOL . PHP_EOL);
}

$sampleData = fopen('../../sample_data', 'r');
$sampleIndex = 1;
while (!feof($sampleData)) {
    $line = fgets($sampleData);
    if (empty(trim($line))) {
        continue;
    }

    $sample = explode(',', $line);
    array_walk($sample, function(&$point) {
        $point = floatval($point);
    });

    calculateMean($sampleIndex, $sample);
    $sampleIndex++;
}

$sample = [];
for ($i = 0; $i < 25; $i++) {
    $sample[] = rand(1, 20);
}
calculateMean($sampleIndex, $sample);